lib/fork.js

```diff
@@ -67,11 +79,23 @@ module.exports = function (args) {
 		ps.emit('data', data);
 	});
 
-	promise.on = function () {
-		ps.on.apply(ps, arguments);
+	var stdout = new Promise(function (resolve) {
+		ps.stdout.on('end', resolve);
+	});
+
+	var stderr = new Promise(function (resolve) {
+		ps.stderr.on('end', resolve);
+	});
 
+	var endPromise = Promise.all([promise.reflect(), stdout, stderr]).then(function () {
```

Comment:
I've come at this at least 5 times, and every time, it has required all these fixes.

I don't know that I've ever applied this one last, so it could be unnecessary. I can pull it out with another commit and see what happens.